typings/styled-components.d.ts

```diff
@@ -0,0 +1,162 @@
+import * as React from "react";
+import { StatelessComponent } from "react";
+
+type ConstrainedProps<C, P> = C & ({ defaultProps?: P } | { new(props?: P, context?: any): any });
+type StyledProps<P> = P & { theme: any };
+
+interface StyledFunction<T, P> {
+  (strs: TemplateStringsArray, ...fns: Array<(props: P) => string>): T;
```

Comment:
I would recommend `fns` to be of type `ReadonlyArray<Interpolation>`.
`export type Interpolation = ((executionContext: StyledProps<P>) => string) | string | number`
- it aligns with flow type
- it allows to pass strings and numbers in addition to functions